### PR TITLE
INT-6696 docker ingress class name

### PR DESCRIPTION
**Pull request by jflinchbaugh**
#### Description of Change

* add the ingressClassName to the docker ingress like it was to the repo ingress.
* some small cleanups
* extra tests around ingress

JIRA: https://issues.sonatype.org/browse/INT-6696
Build: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/Helm3%20Charts/job/Feature%20Snapshot%20Builds/job/INT-6696-docker-ingressClassName/

I'm still putting together a proper plan to test docker repo end-to-end. It is provably listening now, but I need to work around certificates, since it's TLS.

### CODE REVIEW

File: charts/nexus-repository-manager/templates/ingress.yaml [hunk 1]
```diff
@@ -62,6 +62,9 @@ metadata:
       {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
+  {{- if $.Values.ingress.ingressClassName }}
+  ingressClassName: {{ $.Values.ingress.ingressClassName }}
+  {{- end }}
```
Comment (jflinchbaugh) [inline charts/nexus-repository-manager/templates/ingress.yaml:67]: the new fix.

File: charts/nexus-repository-manager/tests/ingress_test.yaml [hunk 2]
```diff
@@ -97,7 +98,22 @@ tests:
...
+              - host: repo.demo
+                http:
+                  paths:
+                    - path: /
+                      pathType: Prefix
+                      backend:
+                        service:
+                          name: RELEASE-NAME-nexus-repository-manager
+                          port:
+                            number: 8081
```
Comment (jflinchbaugh) [inline charts/nexus-repository-manager/tests/ingress_test.yaml:116]: show normal existence of ingressClassName for repo url to contrast for when it's not specified in a later test.

File: charts/nexus-repository-manager/tests/ingress_test.yaml [hunk 3]
```diff
@@ -119,9 +136,90 @@ tests:
                             number: 5000
             tls:
               - hosts:
-                - docker.repo.demo
+                  - docker.repo.demo
```
Comment (jflinchbaugh) [inline charts/nexus-repository-manager/tests/ingress_test.yaml:139]: whitespace/lint fix for the test.

File: charts/nexus-repository-manager/values.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ nexus:
     # registries:
     #   - host: chart.local
     #     port: 5000
-    #     secretName: registrySecret
+    #     secretName: registry-secret
```
Comment (jflinchbaugh) [inline charts/nexus-repository-manager/values.yaml:19]: this sample secret name was in an invalid format (must be lowercase).

**Review by eduard-tita — APPROVED** [review]
+1 LGTM

**Review by hectorhuol — APPROVED** [review]
+1 LGTM

**Pull request merged**